libr/util/str.c

```diff
@@ -1381,6 +1381,23 @@ static char *r_str_escape_(const char *buf, int dot_nl, bool parse_esc_seq, bool
 	return new_buf;
 }
 
+/* hex-escape unprintable characters in a raw buffer (null-safe) */
+R_API char *r_str_escape_raw(const ut8 *buf, int sz) {
+	r_return_val_if_fail (buf, NULL);
+
+	/* Worst case scenario, we convert every byte to a \xhh escape */
+	char *new_buf = malloc (1 + sz * 4);
+	if (!new_buf) {
+		return NULL;
+	}
+	char *q = new_buf;
+	for (int i = 0; i < sz; i++) {
```

Comment:
Move this int outside the for

Reply:
What's wrong with loop scope? Isn't it more confusing to declare outside?

Reply:
It wont compile on macos-ppc and other systems with toolchains before y2k. Its a little change that doesnt hurt much and actually it looks to me like a c++ism, () shouldnt be a scope to define variables imho

Reply:
It's C99, but imo being able to access a loop variable outside is not good and leads to people doing weird things with it, and smaller scope = less clutter when things do get more complicated.

If we need to do this for C89 or other weird legacy compatibility we should note that in developers/contributing docs